backend/services/team_service.py

```diff
@@ -539,3 +541,26 @@ def check_team_membership(project_id: int, allowed_roles: list, user_id: int):
             if TeamService.is_user_an_active_team_member(team_dto.team_id, user_id)
         ]
         return len(user_membership) > 0
+
+    @staticmethod
+    def send_message_to_all_team_members(team_id: int, message_dto: MessageDTO):
+        """  Sends supplied message to all contributors in a team.  Message all team members can take
+             over a minute to run, so this method is expected to be called on its own thread """
+        print(team_id, message_dto)
+        app = (
+            create_app()
+        )  # Because message-all run on background thread it needs it's own app context
+
+        with app.app_context():
+            team_members = TeamService._get_team_members(team_id)
+
+            messages = []
+            for team_member in team_members:
+                print(team_member)
+                message = Message.from_dto(team_member.user_id, message_dto)
+                message.message_type = MessageType.BROADCAST.value
```

Comment:
Instead of using "broadcast" we could create a "Team communication" type

Reply:
Added a new message type